examples/blueprint-ipv6-construct/index.ts

```diff
@@ -0,0 +1,233 @@
+import * as cdk from 'aws-cdk-lib';
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { CapacityType, KubernetesVersion, NodegroupAmiType, IpFamily } from 'aws-cdk-lib/aws-eks';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { Construct } from "constructs";
+import * as blueprints from '../../lib';
+import { logger, userLog } from '../../lib/utils';
+import * as team from '../teams';
+
+const burnhamManifestDir = './examples/teams/team-burnham/';
+const rikerManifestDir = './examples/teams/team-riker/';
+const teamManifestDirList = [burnhamManifestDir, rikerManifestDir];
+const blueprintID = 'blueprint-ipv6-construct-dev';
+
+export interface BlueprintIPv6ConstructProps {
+    /**
+     * Id
+     */
+    id: string
+}
+
+/*
+** This class is modification of ../blueprint-contruct/index.ts
+** To have IPv6 configurations, BlueprintConstruct has been replicated and updated to BlueprintIPV6Construct.
+ */
+export default class BlueprintIPV6Construct {
+    constructor(scope: Construct, props: cdk.StackProps) {
+
+        blueprints.HelmAddOn.validateHelmVersions = true;
+        blueprints.HelmAddOn.failOnVersionValidation = false;
+        logger.settings.minLevel = 3; // info
+        userLog.settings.minLevel = 2; // debug
+
+        const teams: Array<blueprints.Team> = [
+            new team.TeamTroi,
+            new team.TeamRiker(scope, teamManifestDirList[1]),
+            new team.TeamBurnham(scope, teamManifestDirList[0]),
+            new team.TeamPlatform(process.env.CDK_DEFAULT_ACCOUNT!)
+        ];
+
+        const nodeRole = new blueprints.CreateIPv6NodeRoleProvider("blueprint-node-role", new iam.ServicePrincipal("ec2.amazonaws.com"),
```

Comment:
Why do we need to pass principal and managed policies to this provider? Should not that be needed in 90% of cases? If so , why not put that logic in the provider itself and if customers want to pass extra, we can let them.